### ISSUE TITLE
🚀 Feature: Log streaming support in Kubernetes Plugin

### ISSUE THREAD

**Issue by OrkoHunter** [backstage/backstage#25598]
### 🔖 Feature description

The Pod logs viewer in the Kubernetes plugin should stream logs and update the view live. Currently it only fetches the logs for one time on page load, it seems.

![Screenshot 2024-07-12 at 10 08 34 AM](https://github.com/user-attachments/assets/9d015c97-f3a4-4bfa-bed1-e744db223954)
![Screenshot 2024-07-12 at 10 08 47 AM](https://github.com/user-attachments/assets/acee7aa5-70b0-45c7-bd2e-c49fcaf93fa2)

### 🎤 Context

Developers really like the K8S plugin as they start their deployment and watch their new pods spin up. Having the logs viewer stream logs live will help them keep Backstage open in one screen and observe the logs as they deploy or test something on an existing deployment.

### ✌️ Possible Implementation

This can be done similarly to how Scaffolder streams logs from the backend to the UI. The current LogViewer component seems to be making a one time call to fetch logs over here. https://github.com/backstage/backstage/blob/master/plugins/kubernetes-react/src/components/Pods/PodLogs/PodLogs.tsx#L48

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, but I'm happy to collaborate on a PR with someone else

**Comment by jelenazivovic** [comment]
Any update on this?

**Comment by freben** [comment]
No, I don't think it got picked up (@ahhhndre - right?) but feel free to give it a go. 🙏

**Comment by 0xts** [comment]
@freben I'd like to contribute towards this.

**Comment by GersonTrj** [comment]
@freben do you know why this was closed as not planned? I think that having live logs in the kubernetes plugin is quite an important feature to have since static logs are not that useful to check.

**Comment by awanlin** [comment]
Hi @GersonTrj, it's often handy to look at the comment history: on February 10th it was marked as being stale and a week later on February 17 it was close because of that. In both cases this was done by a bot so there isn't any specific reason than the conversation went stale and no one picked up the issue to work on it.

@0xts, are you still interesting in contributing this, have you started any work on it? Feel free to do so.

**Comment by awanlin** [comment]
Thanks for the update @0xts, I've assigned the issue to you 🚀

**Comment by freben** [comment]
@0xts I found this in triaging today - sadly it seems the PR went unnoticed in the end. Sorry about that. Are you still up for reviving the effort? Even if it's in the form of a simple reload every N seconds in the frontend or so?

**Comment by 0xts** [comment]
@freben Actually, I'd raised a PR, but it was never merged. The reason that I was given was that it was too complicated of a change. It's [29995](https://github.com/backstage/backstage/pull/29995). I won't be able to make changes to simplify this to a polling logic, I haven't the time these days, but I can surely rebase it if this implementation seems fine.

**Comment by camilaibs** [comment]
Hi @0xts, I'm sorry about that. I'll add a `contrib-welcome` label to this issue so we can find someone to continue where you left (maybe even you in the future, if you find time 🙂 ). Thanks again for starting to help with this.
